_posts/2021-09-08-deep-dive-into-a-debezium-community-connector-scylla-cdc-source-connector.adoc

```diff
@@ -0,0 +1,229 @@
+
++++<!-- more -->+++
+
+== CDC support in Scylla
+
+Change Data Capture (CDC) allows users to track data modifications in their Scylla database. It can be easily enabled/disabled on any Scylla table. Upon turning it on, a log of all modifications (INSERTs, UPDATEs, DELETEs) will be created and automatically updated.
+
+When we designed our implementation of CDC in Scylla, we wanted to make it easy to consume the CDC log. Therefore, the CDC log is stored as a regular Scylla table, accessible by any existing CQL driver. When a modification is made to a table with CDC enabled, information about that operation is saved to the CDC log table. 
+
+Here’s a quick demo of it: First, we will create a table with CDC enabled:
+
+[source,sql]
+----
+CREATE TABLE ks.orders(
+    user text, 
+    order_id int, 
+    order_name text, 
+    PRIMARY KEY(user, order_id)
+) WITH cdc = {'enabled': true};
+----
+
+Next, let’s perform some operations:
+
+[source,sql]
+----
+INSERT INTO ks.orders(user, order_id, order_name) 
+    VALUES ('Tim', 1, 'apple');
+
+INSERT INTO ks.orders(user, order_id, order_name) 
+    VALUES ('Alice', 2, 'blueberries');
+
+UPDATE ks.orders 
+    SET order_name = 'pineapple' 
+    WHERE user = 'Tim' AND order_id = 1;
+----
+
+Finally, let’s see the contents of the modified table:
+
+[source,sql]
+----
+SELECT * FROM ks.orders;
+
+ user | order_id | order_name
+------+----------+-------------
+  Tim |        1 |   pineapple
+Alice |        2 | blueberries
+----
+
+Looking only at this table, you cannot easily reconstruct all modifications that took place, e.g. Tim’s order name before the UPDATE. Let’s look at the CDC log, easily accessible as a table (some columns truncated for clarity):
+
+[source,sql]
+----
+SELECT * FROM ks.orders_scylla_cdc_log;
+
+ cdc$stream_id | cdc$time |  | cdc$operation | order_id | order_name  |     user
+---------------+----------+-...-+---------------+----------+-------------+----------
+    0x2e46a... |  7604... | ... |             2 |        1 |       apple |      Tim
+    0x2e46a... |  8fdc... | ... |             1 |        1 |   pineapple |      Tim
+    0x41400... |  808e... | ... |             2 |        2 | blueberries |    Alice
+----
+
+All three operations are visible in the CDC log. There are two INSERTs (`cdc$operation = 2`) and one UPDATE (`cdc$operation = 1`). For each operation, its timestamp is also preserved (`cdc$time` column as time UUID).
```

Comment:
How does a UUID represent a time? This might be worth an explanatory sentence, as it's not obvious (not to me, anyways).